### PR TITLE
Add Pull Request Template for UI team

### PR DESCRIPTION
**Pull request by lucferbux**
## Description

Following [github's pull request template instructions](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/creating-a-pull-request-template-for-your-repository), allow custom PR Template for UI contributions.

By default the common PR Template would appear, with a link to the UI one.

## Merge criteria:

- [X] The commits and have meaningful messages; the author will squash them [after approval](https://github.com/opendatahub-io/opendatahub-community/blob/main/contributor-cheatsheet.md#:~:text=Usually%20this%20is%20done%20in%20last%20phase%20of%20a%20PR%20revision) or will ask to merge with squash.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Comment by lucferbux** [comment]
@ederign @tarilabs I've tested this in other repo to make sure it works, however I haven't been able to do that here.
I'm not even sure if this is a good flow, let me know your thoughts, I'm opening the PR to start a discussion.

**Review by tarilabs — COMMENTED** [review]
/lgtm

but how about "merging" the UI-related points into a section in the existing pull request template?

**Comment by lucferbux** [comment]
> but how about "merging" the UI-related points into a section in the existing pull req

Yeah I'm sorry, I'm changing computers and I didn't have signoff enabled by default, thanks

**Comment by Griffin-Sullivan** [comment]
Yeah I would suggest since there's so much overlap with the normal template that we add a UI only section at the bottom for additional criteria.

**Comment by ederign** [comment]
Before reading this thread, I just suggested the same here -> https://github.com/kubeflow/model-registry/pull/298#issuecomment-2310437399 

+1 to have a single template.

File: .github/pull_request_template.md
Comment (tarilabs) [inline .github/pull_request_template.md:16]: Since in upstream we have now the bot, I believe we could rephrase this as:

```suggestion
- [ ] The commits have meaningful messages; the author will squash them after approval or in case of manual merges will ask to merge with squash.
```

File: .github/pull_request_template.md
Comment (tarilabs) [inline .github/pull_request_template.md:23]:
```suggestion
<!--- You can ignore these if you are doing Go Model Registry REST server, MR Python client, manifest, controller, internal logic, etc changes; aka non-UI / visual changes -->
```

what about ^

**Review by tarilabs — COMMENTED** [review]
thanks @lucferbux , see below minor suggestions

File: .github/pull_request_template.md
Comment (lucferbux) [reply .github/pull_request_template.md]: Yeah, I agree with this.

File: .github/pull_request_template.md
Comment (lucferbux) [reply .github/pull_request_template.md]: Yeah, that sounds about right.

**Review by ederign — APPROVED** [review]

**Comment by ederign** [comment]
Thank you @lucferbux !

**Review by tarilabs — COMMENTED** [review]
/lgtm

thanks Lucas, will wait for another to go in then proceeding for merge.

**Review by tarilabs — APPROVED** [review]

**Pull request merged**
